arelle/plugin/validate/ESEF/Util.py

```diff
@@ -51,6 +51,10 @@ def isInEsefTaxonomy(val: ValidateXbrl, modelObject: ModelObject | None) -> bool
     return (any(ns.startswith(esefNsPrefix) for esefNsPrefix in esefTaxonomyNamespaceURIs))
 
 
+def is2022DisclosureSystem(modelXbrl: ModelXbrl) -> bool:
+    return any("2022" in name for name in modelXbrl.modelManager.disclosureSystem.names)
```

Comment:
Suggest replacing this (and the preexisting `is2021DisclosureSystem` function) with a generic get disclosure system year function and using that to see if the year is newer or older than a specfic year.

```python
YEAR_GROUP = "year"
DISCLOSURE_SYSTEM_YEAR_PATTERN = re.compile(rf"esef-(?P<{YEAR_GROUP}>20\d\d)")


def getDisclosureSystemYear(modelXbrl: ModelXbrl) -> int:
    for name in modelXbrl.modelManager.disclosureSystem.names:
        disclosureSystemYear = DISCLOSURE_SYSTEM_YEAR_PATTERN.fullmatch(name)
        if disclosureSystemYear:
            return int(disclosureSystemYear.group(YEAR_GROUP))
    raise ValueError(f"Unable to determine year of ESEF disclosure system matching pattern 'esef-20XX' from {modelXbrl.modelManager.disclosureSystem.names}")

...
if getDisclosureSystemYear(modelXbrl) >= 2023:
    # 2023 logic
else:
    # 2021 and 2022 logic
```